system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -0,0 +1,51 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2024 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#include "elasticSink.hpp"
+#include <cstdio>
+#include "platform.h"
+
+using namespace hpccMetrics;
+
+extern "C" MetricSink* getSinkInstance(const char *name, const IPropertyTree *pSettingsTree)
+{
+    MetricSink *pSink = new ElasticMetricSink(name, pSettingsTree);
+    return pSink;
+}
+
+
+ElasticMetricSink::ElasticMetricSink(const char *name, const IPropertyTree *pSettingsTree) :
+    PeriodicMetricSink(name, "file", pSettingsTree),
+    ignoreZeroMetrics(false)
```

Comment:
style: less error prone to use a default initializer on the member definition instead.